xbmc/platform/darwin/osx/network/NetworkOsx.cpp

```diff
@@ -37,22 +37,41 @@ std::string CNetworkInterfaceOsx::GetCurrentDefaultGateway() const
   std::string result;
 
   FILE* pipe = popen("echo \"show State:/Network/Global/IPv4\" | scutil | grep Router", "r");
-  usleep(100000);
+
   if (pipe)
   {
-    std::string tmpStr;
-    char buffer[256] = {'\0'};
-    if (fread(buffer, sizeof(char), sizeof(buffer), pipe) > 0 && !ferror(pipe))
+    struct pollfd fds[1];
```

Comment:
Why do you need an array of 1 element? You could just use a normal object and pass its address to poll

Reply:
true